### PR TITLE
PR: Add lsp to spyder.completions entry-points

### LINKED ISSUE #14923: Language server does not start in macOS application (master)
### What steps will reproduce the problem?

The language server does not appear to start in the macOS standalone application and command completion does not work.

Also, when running debug mode there does not appear to be any lsp logs produced. Are these in a new location?

### System
* Spyder version: 5.0.0.dev0 1dd460bf55912e851ec36686c8726e250c9853de (latest master)
* Python version: 3.9.1 64-bit
* Qt version: 5.12.10
* PyQt5 version: 5.12.3
* Operating System: Darwin 19.6.0
```
# Mandatory:
applaunchservices >=0.1.7   :  0.2.1 (OK)
atomicwrites >=1.2.0        :  1.4.0 (OK)
chardet >=2.0.0             :  4.0.0 (OK)
cloudpickle >=0.5.0         :  1.6.0 (OK)
cookiecutter >=1.6.0        :  1.7.2 (OK)
diff_match_patch >=20181111 :  20200713 (OK)
intervaltree >=3.0.2        :  3.1.0 (OK)
IPython >=7.6.0             :  7.21.0 (OK)
jedi =0.17.2                :  0.17.2 (OK)
jsonschema >=3.2.0          :  3.2.0 (OK)
keyring >=17.0.0            :  23.0.0 (OK)
nbconvert >=4.0             :  6.0.7 (OK)
numpydoc >=0.6.0            :  1.1.0 (OK)
parso =0.7.0                :  0.7.0 (OK)
pexpect >=4.4.0             :  4.8.0 (OK)
pickleshare >=0.4           :  0.7.5 (OK)
psutil >=5.3                :  5.8.0 (OK)
pygments >=2.0              :  2.8.1 (OK)
pylint >=1.0                :  2.7.2 (OK)
pyls >=0.36.2;<1.0.0        :  v5.0.0a5+145.g1dd460bf5 (OK)
pyls_black >=0.4.6          :  0.4.6 (OK)
pyls_spyder >=0.3.2         :  0.3.2 (OK)
qdarkstyle >=2.8            :  2.8.1 (OK)
qtawesome >=0.5.7           :  1.0.2 (OK)
qtconsole >=5.0.1           :  5.0.2 (OK)
qtpy >=1.5.0                :  1.9.0 (OK)
setuptools >=39.0.0         :  54.1.1 (OK)
sphinx >=0.6.6              :  3.5.2 (OK)
spyder_kernels >=2.0.0dev0  :  2.0.0.dev0 (OK)
textdistance >=4.2.0        :  4.2.1 (OK)
three_merge >=0.1.1         :  0.1.1 (OK)
watchdog >=0.10.3;<2.0.0    :  1.0.2 (OK)
zmq >=17                    :  22.0.3 (OK)

# Optional:
cython >=0.21               :  0.29.22 (OK)
matplotlib >=2.0.0          :  3.3.4 (OK)
numpy >=1.7                 :  1.20.1 (OK)
pandas >=1.1.1              :  1.2.3 (OK)
scipy >=0.17.0              :  1.6.1 (OK)
sympy >=0.7.3               :  1.7.1 (OK)
```

### PR DESCRIPTION
**Pull request by mrclary**
## Description of Changes

The LSP server was not starting for macOS applications, and apparently bootstrap worked around the same issue some time ago by explicitly adding the spyder.completions entry-points in `bootstrap.py`. However, this is not necessary if `lsp` is added in `setup.py`.
* Added `lsp` to `spyder.completions` entry-points; this resolves issue with LSP not starting for macOS application.
* With `lsp` added to entry-points, the entry-point patching in `bootstrap.py` is not needed (in fact causes an error); instead just create Spyder's egg-info

### Issue(s) Resolved

Fixes #14923

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:
@mrclary

### CODE REVIEW

File: bootstrap.py [hunk 1]
```diff
@@ -179,43 +177,9 @@ def remove_pyls_installation():
...
-        'kite': kite
-    }
-}
-
-# Add the fake distribution to the global working_set
-pkg_resources.working_set.add(d, 'spyder')
-
+# Create Spyder egg info
+subprocess.check_output([sys.executable, '-W', 'ignore', 'setup.py',
+                         'develop', '--no-deps'])
```
Comment (ccordoba12) [inline bootstrap.py:182]: This totally defeats the purpose of our bootstrap script, which is not to alter people's envs at all. So please revert this change.

Comment (mrclary) [reply bootstrap.py:182]: done

**Review by mrclary — COMMENTED** [review]
@ccordoba12, actually, I found that this is much cleaner and does not change the environment at all. It just produces `spyder.egg-info` and nothing else (it does not add anything to `pth` or `site-packages`). This should be better anyway, since it will use the actual plugin architecture, rather than trying to duplicate it in `bootstrap.py`

File: bootstrap.py [hunk 2]
```diff
@@ -179,43 +177,8 @@ def remove_pyls_installation():
...
-        'lsp': lsp,
-        'kite': kite
-    }
-}
-
-# Add the fake distribution to the global working_set
-pkg_resources.working_set.add(d, 'spyder')
-
+subprocess.check_output([sys.executable, '-W', 'ignore',
+                         'setup.py', 'egg_info'])
```
Comment (ccordoba12) [inline bootstrap.py:181]: I was waiting to discuss this with @andfoy today. We decided we prefer to maintain the current way to handle completion entry points to avoid touching people's env.

We have put a lot of effort everywhere in the code to do this when Spyder is started with `bootstrap.py`, and this since the current approach is working, we don't see the need to remove it.

So please revert these changes and this PR will be ready.

Comment (ccordoba12) [reply bootstrap.py]: @mrclary, sorry but we need to merge your PR for our next alpha so I had to make these changes myself.

**Review by ccordoba12 — APPROVED** [review]
Thanks a lot @mrclary for your help with this!
